src/source/src/connector_source.rs

```diff
@@ -58,27 +58,19 @@ struct InnerConnectorSourceReader {
     context: SourceContext,
 }
 
-struct InnerConnectorSourceReaderHandle {
-    stop_tx: oneshot::Sender<()>,
-    join_handle: JoinHandle<()>,
-}
-
 /// [`ConnectorSource`] serves as a bridge between external components and streaming or
 /// batch processing. [`ConnectorSource`] introduces schema at this level while
-/// [`SplitReaderImpl`] simply loads raw content from message queue or file system.
+/// [`SplitReaderImpl`] simply loadsF raw content from message queue or file system.
 /// Parallel means that multiple [`InnerConnectorSourceReader`] will run in parallel during the
 /// `next`, so that 0 or more Splits reads can be handled at the Source level.
+#[allow(dead_code)]
 pub struct ConnectorSourceReader {
     pub config: ConnectorProperties,
     pub parser: Arc<SourceParserImpl>,
     pub columns: Vec<SourceColumnDesc>,
 
-    handles: Option<HashMap<SplitId, InnerConnectorSourceReaderHandle>>,
-    message_rx: Receiver<Result<Vec<SourceMessage>>>,
-    // We need to keep this tx, otherwise the channel will return none with 0 inner readers, and we
-    // need to clone this tx when adding new inner readers in the future.
-    message_tx: Sender<Result<Vec<SourceMessage>>>,
-
+    // merge all streams of inner reader into one
+    all_reader_stream: BoxStream<'static, Result<Vec<SourceMessage>>>,
```

Comment:
How about using `type_alias_impl_trait` to avoid this `Box<dyn>`?

Reply:
> Defining the type alias in a submodule may help this. Anyway, since we yield chunks instead of rows, this won't hurt much.

Still an error, I comment it with todo, maybe fater this feature is stable.